cpp/src/gandiva/selection_vector_test.cc

```diff
@@ -102,15 +103,15 @@ TEST_F(TestSelectionVector, TestInt16PopulateFromBitMap) {
   EXPECT_EQ(status.ok(), true) << status.message();
 
   int bitmap_size = RoundUpNumi64(max_slots) * 8;
-  std::unique_ptr<uint8_t> bitmap(new uint8_t[bitmap_size]);
-  memset(bitmap.get(), 0, bitmap_size);
+  std::vector<uint8_t> bitmap(bitmap_size);
+  memset(&bitmap[0], 0, bitmap_size);
```

Comment:
`bitmap.data()` is more idiomatic, though both are ok.

Reply:
Data should be zero initialized in the vector.

Reply:
Oh, right.

Reply:
Which also implies that all memset are useless with this change.

Reply:
I removed them